### ISSUE TITLE
ISE when gracefully stopping IDE while build is in progress

### ISSUE THREAD

**Issue by NumberFourAG** [NumberFour/n4js#65]
IDEBUG-644

```
com.google.inject.ProvisionException: Guice provision errors:

1) Error in custom provider, java.lang.IllegalStateException: The bundle has not been started!
  at org.eclipse.xtext.service.MethodBasedModule.configure(MethodBasedModule.java:74)
  while locating org.eclipse.xtext.ui.resource.IStorage2UriMapper
    for field at org.eclipse.xtext.ui.editor.formatting.PreferenceStoreWhitespaceInformationProvider.storage2UriMapper(Unknown Source)
  while locating org.eclipse.xtext.ui.editor.formatting.PreferenceStoreWhitespaceInformationProvider
  while locating org.eclipse.xtext.formatting.IWhitespaceInformationProvider
    for field at org.eclipse.xtext.formatting.impl.AbstractDeclarativeFormatter.whitespaceInformationProvider(Unknown Source)
  while locating eu.numberfour.ide.ts.formatting.TypesFormatter
  while locating org.eclipse.xtext.formatting.IFormatter
    for field at org.eclipse.xtext.serializer.impl.Serializer.formatter(Unknown Source)
  while locating org.eclipse.xtext.serializer.impl.Serializer
  while locating org.eclipse.xtext.serializer.ISerializer
    for field at org.eclipse.xtext.resource.XtextResource.serializer(Unknown Source)
  while locating org.eclipse.xtext.linking.lazy.LazyLinkingResource
  while locating org.eclipse.xtext.resource.XtextResource

1 error
    at com.google.inject.internal.InjectorImpl$4.get(InjectorImpl.java:987)
    at org.eclipse.xtext.resource.XtextResourceFactory.createResource(XtextResourceFactory.java:20)
    at org.eclipse.emf.ecore.resource.impl.ResourceSetImpl.createResource(ResourceSetImpl.java:434)
    at org.eclipse.emf.ecore.resource.impl.ResourceSetImpl.demandCreateResource(ResourceSetImpl.java:243)
    at org.eclipse.xtext.resource.XtextResourceSet.getResource(XtextResourceSet.java:260)
    at org.eclipse.xtext.resource.SynchronizedXtextResourceSet.getResource(SynchronizedXtextResourceSet.java:25)
    at eu.numberfour.ide.ts.scoping.builtin.ExecutionEnvironmentDescriptor.processResources(ExecutionEnvironmentDescriptor.java:41)
    at eu.numberfour.ide.ts.scoping.builtin.EnumerableScope.createElements(EnumerableScope.java:52)
    at eu.numberfour.ide.ts.scoping.builtin.EnumerableScope.getLocalElementsByName(EnumerableScope.java:78)
    at org.eclipse.xtext.scoping.impl.AbstractScope.getSingleLocalElementByName(AbstractScope.java:112)
    at org.eclipse.xtext.scoping.impl.AbstractScope.getSingleElement(AbstractScope.java:105)
    at eu.numberfour.ide.ts.scoping.builtin.EnumerableScope.getEObjectOrProxy(EnumerableScope.java:119)
    at eu.numberfour.ide.ts.scoping.builtin.BuiltInTypeScope.getObjectType(BuiltInTypeScope.java:338)
    at eu.numberfour.ide.n4js.scoping.members.MemberScopingHelper._members(MemberScopingHelper.java:257)
    at eu.numberfour.ide.n4js.scoping.members.MemberScopingHelper.members(MemberScopingHelper.java:505)
    at eu.numberfour.ide.n4js.scoping.members.MemberScopingHelper._members(MemberScopingHelper.java:402)
    at eu.numberfour.ide.n4js.scoping.members.MemberScopingHelper.members(MemberScopingHelper.java:485)
    at eu.numberfour.ide.n4js.scoping.members.MemberScopingHelper.createMemberScopeFor(MemberScopingHelper.java:106)
    at eu.numberfour.ide.n4js.scoping.N4JSScopeProvider.scope_PropertyAccessExpression_property(N4JSScopeProvider.java:496)
    at eu.numberfour.ide.n4js.scoping.N4JSScopeProvider.getScope(N4JSScopeProvider.java:195)
    at org.eclipse.xtext.linking.impl.DefaultLinkingService.getScope(DefaultLinkingService.java:59)
    at eu.numberfour.ide.n4js.resource.ErrorAwareLinkingService.getLinkedObjects(ErrorAwareLinkingService.java:53)
    at org.eclipse.xtext.linking.lazy.LazyLinkingResource.getEObject(LazyLinkingResource.java:250)
    at org.eclipse.xtext.linking.lazy.LazyLinkingResource.getEObject(LazyLinkingResource.java:225)
    at org.eclipse.xtext.resource.persistence.StorageAwareResource.getEObject(StorageAwareResource.java:136)
    at org.eclipse.emf.ecore.resource.impl.ResourceSetImpl.getEObject(ResourceSetImpl.java:223)
    at org.eclipse.emf.ecore.util.EcoreUtil.resolve(EcoreUtil.java:203)
    at org.eclipse.emf.ecore.util.EcoreUtil.resolve(EcoreUtil.java:259)
    at org.eclipse.emf.ecore.impl.BasicEObjectImpl.eResolveProxy(BasicEObjectImpl.java:1473)
    at eu.numberfour.ide.n4js.n4JS.impl.ParameterizedPropertyAccessExpressionImpl.getProperty(ParameterizedPropertyAccessExpressionImpl.java:161)
    at eu.numberfour.ide.n4js.n4JS.impl.ParameterizedPropertyAccessExpressionImpl.eGet(ParameterizedPropertyAccessExpressionImpl.java:231)
    at org.eclipse.emf.ecore.impl.BasicEObjectImpl.eGet(BasicEObjectImpl.java:1011)
    at org.eclipse.emf.ecore.impl.BasicEObjectImpl.eGet(BasicEObjectImpl.java:1003)
    at eu.numberfour.ide.n4js.postprocessing.ASTProcessor.resolveProxiesInNode(ASTProcessor.java:420)
    at eu.numberfour.ide.n4js.postprocessing.ASTProcessor.processSubtree(ASTProcessor.java:232)
    at eu.numberfour.ide.n4js.postprocessing.ASTProcessor.processSubtree(ASTProcessor.java:224)
    at eu.numberfour.ide.n4js.postprocessing.ASTProcessor.processSubtree(ASTProcessor.java:224)
    at eu.numberfour.ide.n4js.postprocessing.ASTProcessor.processSubtree(ASTProcessor.java:224)
    at eu.numberfour.ide.n4js.postprocessing.ASTProcessor.processSubtree(ASTProcessor.java:224)
    at eu.numberfour.ide.n4js.postprocessing.ASTProcessor.processSubtree(ASTProcessor.java:224)
    at eu.numberfour.ide.n4js.postprocessing.ASTProcessor.processSubtree(ASTProcessor.java:224)
    at eu.numberfour.ide.n4js.postprocessing.ASTProcessor.processSubtree(ASTProcessor.java:224)
    at eu.numberfour.ide.n4js.postprocessing.ASTProcessor.processAST(ASTProcessor.java:124)
    at eu.numberfour.ide.n4js.postprocessing.ASTProcessor.processAST(ASTProcessor.java:99)
    at eu.numberfour.ide.n4js.resource.N4JSPostProcessor.performPostProcessing(N4JSPostProcessor.java:54)
    at eu.numberfour.ide.n4js.resource.PostProcessingAwareResource.performPostProcessing(PostProcessingAwareResource.java:166)
    at eu.numberfour.ide.n4js.resource.N4JSResource.performPostProcessing(N4JSResource.java:726)
    at eu.numberfour.ide.n4js.resource.PostProcessingAwareResource.resolveLazyCrossReferences(PostProcessingAwareResource.java:118)
    at eu.numberfour.ide.n4js.resource.N4JSResource.resolveLazyCrossReferences(N4JSResource.java:735)
    at org.eclipse.xtext.EcoreUtil2.resolveLazyCrossReferences(EcoreUtil2.java:528)
    at eu.numberfour.ide.n4js.ui.building.ClusteringBuilderState_MODIFIED.indexResource(ClusteringBuilderState_MODIFIED.java:308)
    at eu.numberfour.ide.n4js.ui.building.ClusteringBuilderState_MODIFIED.doUpdate(ClusteringBuilderState_MODIFIED.java:202)
    at eu.numberfour.ide.n4js.ui.building.N4JSGenerateImmediatelyBuilderState.doUpdate(N4JSGenerateImmediatelyBuilderState.java:84)
    at org.eclipse.xtext.builder.builderState.AbstractBuilderState.update(AbstractBuilderState.java:115)
    at org.eclipse.xtext.builder.impl.XtextBuilder.doBuild(XtextBuilder.java:248)
    at eu.numberfour.ide.n4js.ui.building.N4JSBuildTypeTrackingBuilder.doBuild(N4JSBuildTypeTrackingBuilder.java:47)
    at org.eclipse.xtext.builder.impl.XtextBuilder.incrementalBuild(XtextBuilder.java:228)
    at org.eclipse.xtext.builder.impl.XtextBuilder.build(XtextBuilder.java:123)
    at org.eclipse.core.internal.events.BuildManager$2.run(BuildManager.java:734)
    at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:42)
    at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:205)
    at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:245)
    at org.eclipse.core.internal.events.BuildManager$1.run(BuildManager.java:300)
    at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:42)
    at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:303)
    at org.eclipse.core.internal.events.BuildManager.basicBuildLoop(BuildManager.java:359)
    at org.eclipse.core.internal.events.BuildManager.build(BuildManager.java:382)
    at org.eclipse.core.internal.events.AutoBuildJob.doBuild(AutoBuildJob.java:144)
    at org.eclipse.core.internal.events.AutoBuildJob.run(AutoBuildJob.java:235)
    at org.eclipse.core.internal.jobs.Worker.run(Worker.java:55)
Caused by: java.lang.IllegalStateException: The bundle has not been started!
    at org.eclipse.xtext.ui.shared.Access$InternalProvider.get(Access.java:45)
    at com.google.inject.internal.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:40)
    at com.google.inject.internal.SingleFieldInjector.inject(SingleFieldInjector.java:53)
    at com.google.inject.internal.MembersInjectorImpl.injectMembers(MembersInjectorImpl.java:110)
    at com.google.inject.internal.ConstructorInjector.construct(ConstructorInjector.java:94)
    at com.google.inject.internal.ConstructorBindingImpl$Factory.get(ConstructorBindingImpl.java:254)
    at com.google.inject.internal.FactoryProxy.get(FactoryProxy.java:54)
    at com.google.inject.internal.SingleFieldInjector.inject(SingleFieldInjector.java:53)
    at com.google.inject.internal.MembersInjectorImpl.injectMembers(MembersInjectorImpl.java:110)
    at com.google.inject.internal.ConstructorInjector.construct(ConstructorInjector.java:94)
    at com.google.inject.internal.ConstructorBindingImpl$Factory.get(ConstructorBindingImpl.java:254)
    at com.google.inject.internal.FactoryProxy.get(FactoryProxy.java:54)
    at com.google.inject.internal.SingleFieldInjector.inject(SingleFieldInjector.java:53)
    at com.google.inject.internal.MembersInjectorImpl.injectMembers(MembersInjectorImpl.java:110)
    at com.google.inject.internal.ConstructorInjector.construct(ConstructorInjector.java:94)
    at com.google.inject.internal.ConstructorBindingImpl$Factory.get(ConstructorBindingImpl.java:254)
    at com.google.inject.internal.FactoryProxy.get(FactoryProxy.java:54)
    at com.google.inject.internal.SingleFieldInjector.inject(SingleFieldInjector.java:53)
    at com.google.inject.internal.MembersInjectorImpl.injectMembers(MembersInjectorImpl.java:110)
    at com.google.inject.internal.ConstructorInjector.construct(ConstructorInjector.java:94)
    at com.google.inject.internal.ConstructorBindingImpl$Factory.get(ConstructorBindingImpl.java:254)
    at com.google.inject.internal.FactoryProxy.get(FactoryProxy.java:54)
    at com.google.inject.internal.InjectorImpl$4$1.call(InjectorImpl.java:978)
    at com.google.inject.internal.InjectorImpl.callInContext(InjectorImpl.java:1024)
    at com.google.inject.internal.InjectorImpl$4.get(InjectorImpl.java:974)
    ... 69 more
```

```
java.lang.IllegalStateException: cache miss: no actual type in cache for AST node: eu.numberfour.ide.n4js.n4JS.impl.IdentifierRefImpl@2b6640e6 in resource: platform:/resource/eu.numberfour.stdlib.model.base.tests/test/n4js/n4/util/PSParserPrivateMethodsTest.n4js
    at eu.numberfour.ide.n4js.postprocessing.TypingCacheHelper$TypingCache.get(TypingCacheHelper.java:72)
    at eu.numberfour.ide.n4js.postprocessing.TypeProcessor.xsemantics_type(TypeProcessor.java:272)
    at eu.numberfour.ide.n4js.typesystem.CustomTypeSystem.typeInternal(CustomTypeSystem.java:165)
    at eu.numberfour.ide.n4js.xsemantics.N4JSTypeSystem.type(N4JSTypeSystem.java:578)
    at eu.numberfour.ide.n4js.xsemantics.N4JSTypeSystem.type(N4JSTypeSystem.java:573)
    at eu.numberfour.ide.n4js.typeinference.N4JSTypeInferencer.tau(N4JSTypeInferencer.java:147)
    at eu.numberfour.ide.n4js.typeinference.N4JSTypeInferencer.tau(N4JSTypeInferencer.java:91)
    at eu.numberfour.ide.n4js.typeinference.N4JSTypeInferencer.tau(N4JSTypeInferencer.java:83)
    at eu.numberfour.ide.n4js.scoping.N4JSScopeProvider.scope_PropertyAccessExpression_property(N4JSScopeProvider.java:491)
    at eu.numberfour.ide.n4js.scoping.N4JSScopeProvider.getScope(N4JSScopeProvider.java:195)
    at org.eclipse.xtext.linking.impl.DefaultLinkingService.getScope(DefaultLinkingService.java:59)
    at eu.numberfour.ide.n4js.resource.ErrorAwareLinkingService.getLinkedObjects(ErrorAwareLinkingService.java:53)
    at org.eclipse.xtext.linking.lazy.LazyLinkingResource.getEObject(LazyLinkingResource.java:250)
    at org.eclipse.xtext.linking.lazy.LazyLinkingResource.getEObject(LazyLinkingResource.java:225)
    at org.eclipse.xtext.resource.persistence.StorageAwareResource.getEObject(StorageAwareResource.java:136)
    at org.eclipse.emf.ecore.resource.impl.ResourceSetImpl.getEObject(ResourceSetImpl.java:223)
    at org.eclipse.emf.ecore.util.EcoreUtil.resolve(EcoreUtil.java:199)
    at org.eclipse.emf.ecore.util.EcoreUtil.resolve(EcoreUtil.java:259)
    at org.eclipse.emf.ecore.impl.BasicEObjectImpl.eResolveProxy(BasicEObjectImpl.java:1473)
    at eu.numberfour.ide.n4js.n4JS.impl.ParameterizedPropertyAccessExpressionImpl.getProperty(ParameterizedPropertyAccessExpressionImpl.java:161)
    at eu.numberfour.ide.n4js.n4JS.impl.ParameterizedPropertyAccessExpressionImpl.eGet(ParameterizedPropertyAccessExpressionImpl.java:231)
    at org.eclipse.emf.ecore.impl.BasicEObjectImpl.eGet(BasicEObjectImpl.java:1011)
    at org.eclipse.emf.ecore.impl.BasicEObjectImpl.eGet(BasicEObjectImpl.java:1003)
    at eu.numberfour.ide.n4js.resource.N4JSCrossReferenceComputer.handleReference(N4JSCrossReferenceComputer.java:84)
    at eu.numberfour.ide.n4js.resource.N4JSCrossReferenceComputer.computeCrossRefs(N4JSCrossReferenceComputer.java:73)
    at eu.numberfour.ide.n4js.resource.N4JSCrossReferenceComputer.computeCrossRefs(N4JSCrossReferenceComputer.java:59)
    at eu.numberfour.ide.n4js.resource.N4JSResourceDescription.getImportedNames(N4JSResourceDescription.java:82)
    at org.eclipse.xtext.builder.builderState.BuilderStateUtil.copyImportedNames(BuilderStateUtil.java:40)
    at org.eclipse.xtext.builder.builderState.BuilderStateUtil.create(BuilderStateUtil.java:35)
    at eu.numberfour.ide.n4js.ui.building.ClusteringBuilderState_MODIFIED.doUpdate(ClusteringBuilderState_MODIFIED.java:207)
    at eu.numberfour.ide.n4js.ui.building.N4JSGenerateImmediatelyBuilderState.doUpdate(N4JSGenerateImmediatelyBuilderState.java:84)
    at org.eclipse.xtext.builder.builderState.AbstractBuilderState.update(AbstractBuilderState.java:115)
    at org.eclipse.xtext.builder.impl.XtextBuilder.doBuild(XtextBuilder.java:248)
    at eu.numberfour.ide.n4js.ui.building.N4JSBuildTypeTrackingBuilder.doBuild(N4JSBuildTypeTrackingBuilder.java:47)
    at org.eclipse.xtext.builder.impl.XtextBuilder.incrementalBuild(XtextBuilder.java:228)
    at org.eclipse.xtext.builder.impl.XtextBuilder.build(XtextBuilder.java:123)
    at org.eclipse.core.internal.events.BuildManager$2.run(BuildManager.java:734)
    at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:42)
    at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:205)
    at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:245)
    at org.eclipse.core.internal.events.BuildManager$1.run(BuildManager.java:300)
    at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:42)
    at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:303)
    at org.eclipse.core.internal.events.BuildManager.basicBuildLoop(BuildManager.java:359)
    at org.eclipse.core.internal.events.BuildManager.build(BuildManager.java:382)
    at org.eclipse.core.internal.events.AutoBuildJob.doBuild(AutoBuildJob.java:144)
    at org.eclipse.core.internal.events.AutoBuildJob.run(AutoBuildJob.java:235)
    at org.eclipse.core.internal.jobs.Worker.run(Worker.java:55)
```

**Comment by mor-n4** [comment]
this will probably be resolved once issues #228 and #225 are done

**Comment by mor-n4** [comment]
considering the above stack trace, this should be fixed by #225 (and also #228);
since the issue description does not state how to reproduce the problem, closing this now
